### PR TITLE
Implement Talk recording

### PR DESCRIPTION
**Pull request by enoch85**
Fix https://github.com/nextcloud/vm/issues/2519

cc @szaimen

### CODE REVIEW

File: apps/talk.sh [hunk 1]
```diff
@@ -486,3 +486,35 @@ else
...
+shm_size=2g \
+-e NC_DOMAIN=${TURN_DOMAIN} \
+-e TZ=${TIMEZONE} \
+-e RECORDING_SECRET=${TURN_RECORDING_SECRET} \
+-e INTERNAL_SECRET=${TURN_INTERNAL_SECRET}
+
+# Talk recording
+if [ -d "$NCPATH/apps/spreed" ]
+then
+    if [ "$TALK_RECORDING_ENABLED" = 'yes' ]
```
[the remark this answers is not in the document]
Comment (enoch85) [reply apps/talk.sh:509]: @szaimen Never found the check you run here, but I guess it's something like `is_this_docker_running talk_recording`?

File: apps/talk.sh [hunk 2]
```diff
@@ -486,3 +488,33 @@ else
     msg_box "Congratulations, everything is working as intended! The installation succeeded.\n\nLogging can be found by typing: journalctl -lfu signaling"
     exit 0
 fi
+
+####### Talk recording
+print_text_in_color "$ICyan" "Setting up Talk recording..."
```
Comment (szaimen) [inline apps/talk.sh:493]: I would recommend to make this optional as talk recording is very resource intensive.

Comment (enoch85) [reply apps/talk.sh:493]: Yeah sure, been thinking about that too.

File: apps/talk.sh [hunk 3]
```diff
@@ -325,7 +327,7 @@ debug = false
 hashkey = $(openssl rand -hex 16)
 blockkey = $(openssl rand -hex 16)
 [clients]
-internalsecret = $(openssl rand -hex 16)
+internalsecret = $(TURN_INTERNAL_SECRET)
```
Comment (szaimen) [inline apps/talk.sh:330]:
```suggestion
internalsecret = ${TURN_INTERNAL_SECRET}
```

**Comment by enoch85** [comment]
@szaimen What do you think? Did I understand you correctly or is something wrong here?

Not tested.

File: apps/talk.sh [hunk 4]
```diff
@@ -483,6 +485,43 @@ then
...
+
+# It's pretty recource intensive
+cpu_check 4
+
+print_text_in_color "$ICyan" "Setting up Talk recording..."
+
+# Pull and start
+docker pull nextcloud/aio-talk-recording:latest
+docker run -t -d -p "$TURN_RECORDING_HOST":"$TURN_RECORDING_HOST_PORT":"$TURN_RECORDING_HOST_PORT" \
+--restart \
```
Comment (szaimen) [inline apps/talk.sh:505]:
```suggestion
--restart always \
```

File: apps/talk.sh [hunk 5]
```diff
@@ -483,6 +485,43 @@ then
...
+--name talk-recording nextcloud/aio-talk-recording \
```
Comment (szaimen) [inline apps/talk.sh:506]:
```suggestion
--name talk-recording \
```

**Comment by szaimen** [comment]
> @szaimen What do you think? Did I understand you correctly or is something wrong here?
> 
> Not tested.

lgtm in general. However If you compare the talk-hpb with https://github.com/nextcloud/all-in-one/blob/main/Containers/talk/start.sh, the turn section can be removed

**Comment by enoch85** [comment]
> However If you compare the talk-hpb with https://github.com/nextcloud/all-in-one/blob/main/Containers/talk/start.sh, the turn section can be removed

Hmm, not so sure about that. I've seen it in an enterprise setup as well. As you already know, we use @morph027 setup for Signaling, and the `JANUS_KEY` is there, not added by me, and it works as is. This is just an addition.

I suppose you mean this: https://github.com/nextcloud/vm/blob/master/apps/talk.sh#L341-L343

**Comment by szaimen** [comment]
> I suppose you mean this: https://github.com/nextcloud/vm/blob/master/apps/talk.sh#L341-L343

yes

**Comment by enoch85** [comment]
> > I suppose you mean this: https://github.com/nextcloud/vm/blob/master/apps/talk.sh#L341-L343
> 
> yes

I'll leave it, don't know the side effects if I remove it.

**Comment by szaimen** [comment]
> I'll leave it, don't know the side effects if I remove it.

The side-effect of leaving it could be that the recording server does not work.

**Comment by enoch85** [comment]
> > I'll leave it, don't know the side effects if I remove it.
> 
> The side-effect of leaving it could be that the recording server does not work.

Actually, we have that in another setup (not your container, and my colleges installation), and it's working there. Do you mean that your container doesn't handle it, or what's the reason for it stop working?

Also, it should only be removed if talk recording is turned on?

**Comment by szaimen** [comment]
> Actually, we have that in another setup (not your container, and my colleges installation), and it's working there. Do you mean that your container doesn't handle it, or what's the reason for it stop working?
> 
> Also, it should only be removed if talk recording is turned on?

The config is simply wrong as far as I know. The talk-hpb does not need a turn server afaik. Turn is only used by the android and ios apps directly.

**Comment by enoch85** [comment]
> The config is simply wrong as far as I know.

Can you give your take on this?

I will merge this now for easier testing. We can continue the discussion in this PR though.

**Pull request merged**